rust/agents/validator/src/validator.rs

```diff
@@ -268,24 +260,33 @@ impl Validator {
                     announced_locations=?locations,
                     "Validator has not announced signature storage location"
                 );
-                let balance_delta = self
-                    .validator_announce
-                    .announce_tokens_needed(signed_announcement.clone())
-                    .await
-                    .unwrap_or_default();
-                if balance_delta > U256::zero() {
-                    warn!(
-                        tokens_needed=%balance_delta,
-                        validator_address=?announcement.validator,
-                        "Please send tokens to the validator address to announce",
-                    );
-                } else {
-                    let result = self
+
+                if self.core.settings.chains[self.origin_chain.name()]
```

Comment:
The commit that's running doesn't actually include this change. However I noticed that the fix that was added recently would just not even create the announcement.json if you don't have a signer set, which we don't want. Instead we want to still create an announcement.json and loudly complain while the validator still isn't announced